### ISSUE TITLE
[Bug]: Very poor accessibility – literally unusable

### ISSUE THREAD

**Issue by reza-sadeghzadeh** [mattermost/mattermost#36489]
### Before you file a bug report

- [x] I have checked the [issue tracker](https://github.com/mattermost/mattermost/issues) and have not found an issue that matches the one I'm filing.
- [x] This issue is not a troubleshooting question. Troubleshooting questions go here: https://forum.mattermost.com/c/trouble-shoot/16.
- [x] This issue is not a feature request. You can request features and make product suggestions here: https://mattermost.uservoice.com/forums/306457-general/.
- [x] This issue reproduces on one of the [currently supported server versions](https://docs.mattermost.com/about/mattermost-server-releases.html#latest-releases).
- [x] I have read the [contribution guidelines](https://github.com/mattermost/mattermost/blob/master/CONTRIBUTING.md) and the [Mattermost Handbook Contribution Guidelines](https://handbook.mattermost.com/contributors/contributors/guidelines/contribution-guidelines).

### Mattermost Server Version

Latest

### Operating System

Mac

### Steps to reproduce

I have a teammate who uses visual accessibility tools. He is a true professional in tech and works as an AI engineer. I’ve seen him using other tools like a pro, and he has no issues whatsoever. Yet, he can’t (and doesn’t) use Mattermost due to a lack of accessibility features. I’m in no position to understand these issues or help in any way, I just thought this might be useful feedback for you great people at Mattermost.

### Expected behavior

Mattermost should work seamlessly with common visual accessibility tools.

### Observed behavior

My teammate, who is a skilled AI engineer and uses visual accessibility tools without issue in other professional software, is completely unable to use Mattermost. The lack of accessibility support makes it unusable for him, so he avoids it entirely.

### Log Output

```shell

```

**Comment by realInt2k** [comment]
Why isn't your teammate the one issuing this "Bug"? 
What is the visual accessibility tools he's using?
What features are lacking that he needs?
If he is a true professional in tech, why couldn't he figure a way out himself?
Why being an AI engineer important in this convesation? Are you implying AI engineer is top-tier and smarter than other dev professions?
